Realm/Tests/ObjectTests.m

```diff
@@ -683,6 +683,19 @@ - (void)testStringSizeLimits {
     [realm commitWriteTransaction];
 }
 
+- (void)testAddingObjectNotInSchemaThrows {
+    RLMRealmConfiguration *configuration = [RLMRealmConfiguration defaultConfiguration];
+    configuration.objectClasses = @[StringObject.class];
+    RLMRealm *realm = [RLMRealm realmWithConfiguration:configuration error:nil];
+
+    [realm beginWriteTransaction];
+    RLMAssertThrowsWithReasonMatching([realm addObject:[[IntObject alloc] initWithValue:@[@1]]], @"Object type 'IntObject' not persisted in Realm");
```

Comment:
That case could also use a more specific error message saying that it's not an RLMObject subclass. Can check for isKindOfClass:RLMObjectBase.class in the error case.